### DISCUSSION TITLE
Keybinding Special Characters

### DISCUSSION THREAD

**Discussion by joshmedeski** [ghostty-org/ghostty#2644] [Q&A] [answered]
@mitchellh can you confirm how to get this working?

```
keybind = super+}=text:\x02\x6e
keybind = super+{=text:\x02\x70
```

Not sure if special characters like `{` are not working for a similar issue or I'm missing something.

Perhaps it's related to https://github.com/ghostty-org/ghostty/issues/1802

**Comment by mitchellh** [comment, marked as answer]
Please use the Q&A section in discussions for Q&A. In this case: are `}` and `{` unshifted characters on your keyboard layout? On a US keyboard, you would have to do this: `super+shift+left_bracket` (since shift + left bracket maps to `{` on a US keyboard layout). Depending on your layout this will be different.

**Comment by mitchellh** [comment]
I'll add that you can use the terminal inspector to also see what the raw key events are for the keys that you press. That can help determine how to create bindings.

**Comment by joshmedeski** [comment]
Thanks for moving this to discussions and this code works!

```
keybind = super+shift+left_bracket=text:\x02\x6e
keybind = super+shift+right_bracket=text:\x02\x70
```

Where can I find a list of all special characters available?

**Reply by mitchellh** [reply]
They're currently all in `src/input/key.zig` in an enum. They're not yet documented anywhere else but its a good idea to do so.
